app/uk/gov/hmrc/teamsandrepositories/persistence/RepositoriesPersistence.scala

```diff
@@ -55,14 +55,14 @@ class RepositoriesPersistence @Inject()(mongoComponent: MongoComponent)(implicit
         repoType.map(rt => Filters.equal("repoType", rt.asString)),
     ).flatten
     filters match {
-      case Nil  => collection.find().toFuture()
-      case more => collection.find(Filters.and(more:_*)).toFuture()
+      case Nil  => collection.find().map(GitRepository.buildPrototypeUrl).toFuture()
```

Comment:
Can we add the `prototypeUrl` before inserting into mongo? This will avoid having to add it on each read, and will allow us to provide overrides more easily later.
I suspect `GhRepository.toGitRepository` in `GithubConnector` maybe a good place. This should allow us to take cues from application.conf/repository.yaml etc.

Reply:
I also see that `RepositoryDetails.create` uses url-templates (from application.conf) to populate the environments field. I wonder if we should move the prototype url template out of code into `url-templates` too, and populate in the same place.